### PR TITLE
Small config fixes

### PR DESCRIPTION
**Pull request by zsarnett**
## Proposed change

Removed Header in System Navigation Card
Edit Home Location when in Narrow Mode
Add Search to System Navigation
Make back button color consistent

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/core/ha-config-system-navigation.ts [hunk 1]
```diff
@@ -41,33 +42,31 @@ class HaConfigSystemNavigation extends LitElement {
...
-          @action=${this._handleAction}
-        >
+        <div slot="toolbar-icon">
           <ha-icon-button
-            slot="trigger"
-            .label=${this.hass.localize("ui.common.overflow_menu")}
-            .path=${mdiDotsVertical}
+            .label=${this.hass.localize("ui.dialogs.quick-bar.title")}
+            .path=${mdiMagnify}
+            @click=${this._showQuickBar}
```
Comment (balloob) [inline src/panels/config/core/ha-config-system-navigation.ts:49]: I don't think it adds much value to add it here as there are no further sub-navigation pages. All pages you can search for are the names of the things already on your screen.

Comment (zsarnett) [reply src/panels/config/core/ha-config-system-navigation.ts:49]: Its more for a continuation of the config menu

Comment (zsarnett) [reply src/panels/config/core/ha-config-system-navigation.ts:49]: Sure. I feel like adding the search more places is only adding functionality without much negative 🤷 

But down to remove. I don't have that strong of an opinion

Comment (balloob) [reply src/panels/config/core/ha-config-system-navigation.ts:49]: Let's remove it and only leave it on the entry levels of panels (Lovelace/Config)

Comment (zsarnett) [reply src/panels/config/core/ha-config-system-navigation.ts]: done

**Review by balloob — APPROVED** [review]

**Pull request merged**